pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -264,6 +260,121 @@ func (r *ReconcileMachineHealthCheck) Reconcile(ctx context.Context, request rec
 	return reconcile.Result{}, nil
 }
 
+//TODO (mshitrit): use better func name
+func (r *ReconcileMachineHealthCheck) remediate(ctx context.Context, needRemediationTargets []target, errList []error, m *mapiv1.MachineHealthCheck) []error {
+	// remediate
+	for _, t := range needRemediationTargets {
+		condition := conditions.Get(&t.Machine, mapiv1.MachineHealthCheckSucceededCondition)
+		//TODO(mshitrit): should we consider pause logic ? (was not part of PR)
+		/*if annotations.IsPaused(cluster, t.Machine) {
+			logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+
+		} else {*/
+		if m.Spec.RemediationTemplate != nil {
+			errList = r.externalRemediation(ctx, m, t, errList, condition)
+
+		} else {
+			//TODO(mshitrit) - is this code needed ?
+			//TODO(mshitrit) consider location
+			//TODO(mshitrit) - is condition valid to be nil
+			//klog.V(3).Infof("Target has failed health check, marking for remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+
+			// NOTE: MHC is responsible for creating MachineOwnerRemediatedCondition if missing or to trigger another remediation if the previous one is completed;
+			// instead, if a remediation is in already progress, the remediation owner is responsible for completing the process and MHC should not overwrite the condition.
+			if !conditions.Has(&t.Machine, mapiv1.MachineOwnerRemediatedCondition) || conditions.IsTrue(&t.Machine, mapiv1.MachineOwnerRemediatedCondition) {
+				conditions.MarkFalse(&t.Machine, mapiv1.MachineOwnerRemediatedCondition, mapiv1.WaitingForRemediationReason, mapiv1.ConditionSeverityWarning, "")
+			}
+
+
+			//Prev Code
+			klog.V(3).Infof("Reconciling %s: meet unhealthy criteria, triggers remediation", t.string())
```

Comment:
any reason this message is used only when there's no remediation template? why not in both cases?